addons/stock/models/stock_picking.py

```diff
@@ -647,6 +647,11 @@ def onchange_picking_type(self):
                     'message': partner.picking_warn_msg
                 }}
 
+    @api.onchange('partner_id', 'move_ids_without_package')
+    def onchange_move_partner_id(self):
+        for picking in self:
+            picking.move_ids_without_package.filtered(lambda move: not move.scrapped).partner_id = picking.partner_id.id
+
```

Comment:
Is it possible to modify `onchange_partner_id` instead of creating a new onchange? I think the issue is only occurring when it's a new record because of how the onchanges work. Ideally it would be a compute, but I don't want to potentially mess up the dropshipping flows.